source/server/http/admin.cc

```diff
@@ -395,28 +395,31 @@ Http::Code AdminImpl::handlerStats(const std::string& url, Http::HeaderMap& resp
   return rc;
 }
 
-std::string AdminImpl::sanitizePrometheusName(const std::string& name) {
```

Comment:
I think a more descriptive name than "strict" would be better.  I think the distinction here is whether a stat name or a label name is being sanitized, right?  If so, can you rename to something that implies that?  Also, please comment on what the prometheus rules are, or link to the prometheus docs, so somebody reading the code can understand what's going on here.

Reply:
Actually, I forgot a relevant doubt here.  

If you check the code in ::formatTagsForPrometheus, it uses the sanitize function *both* for the tag name and the value. This seems odd to me, as we should not need to apply the sanitization to the tag value as well. In fact, the change required in this issue needs us *not* to apply the replace on '-'.  I added the "isStrict" to distinguish between real names or values (this is what the 3rd commit contains.)

But, as I said, sanitizing names seems odd and I was wondering if rather than adding that if case, we should just not sanitize values at all.  That is: 

```
+    buf.push_back(fmt::format("{}=\"{}\"", sanitizePrometheusName(tag.name_), tag.value_)); 
```

I'm going to submit a commit addressing your current comments (better naming and comments) for now, but let me know if you agree on my suggestion above and I'll add it later. 

cc @mattklein123

Reply:
Actually, it seems like we should only sanitize the value, and not the name.  Typically, the value will come from configuration (although in the case of cipher suites that isn't true; they come from the ssl library).  Whereas the tag name should be a compile-time constant in envoy somewhere.

Or am I misunderstanding?

Reply:
Comments addressed @ ff21df3

I've added comments to the header file, cleaned up var / method names, and removed the strict hack as looking at Prometheus docs, it seems pointless. According to the [data model](https://prometheus.io/docs/concepts/data_model/), label values may take any unicode character, so there should be no need for sanitization. 

You're right that the name may not need sanitization as it comes from configuration (in our case, the label name comes from [well_known_names.h](https://github.com/envoyproxy/envoy/pull/2303/files#diff-37270bd85c277c4837361a4b59f89b23R210)). I kept it because it's very easy that someone adds tag name unaware that some chars are not supported by Prometheus, and detecting this will be hard. Keeping the label name sanitization would solve that transparently. But let me know if you prefer to remove it. 

Thanks again for the quick reviews & comments.